src/whip-client.c

```diff
@@ -349,6 +371,15 @@ static gboolean whip_initialize(void) {
 		goto err;
 	}
 
+	if(eos_sink_name != NULL) {
+		GstElement* src = gst_bin_get_by_name(GST_BIN(pipeline), eos_sink_name);
+		GstPad* sinkpad = gst_element_get_static_pad(src, "sink");
+		gst_pad_set_event_function(sinkpad, source_events);
+		gst_object_unref(sinkpad);
+	}
+
+	g_object_set(pipeline, "message-forward", TRUE, NULL);
```

Comment:
If this is related to your additions, why not just call it if `eos_sink_name` is not `NULL`? If it is, we're not going to do anything about an EOS anyway.

Reply:
removed, it was part of the old attempt.